### PR TITLE
Add a new wrapper for `point2point` IPAM that filters invalid addresses and routes

### PR DESCRIPTION
**Pull request by NikitaSkrynnik**
## Description

If `point2point` IPAM receives a request with filled `ip_context` it doesn't check if some of `ip_context`'s addresses are already allocated for other connections. The PR adds a new wrapper for `point2point` IPAM that filters already allocated addresses.

## Issue link
https://github.com/networkservicemesh/deployments-k8s/issues/11787

## How Has This Been Tested?

- [x] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [x] New functionality
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/ipam/filteripam/server.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+type filterIPAMServer struct {
+	ipPool *ippool.IPPool
+}
+
+// NewServer - creates a new filter IPAM server
+func NewServer(newIPAMServer func(...*net.IPNet) networkservice.NetworkServiceServer, prefixes ...*net.IPNet) networkservice.NetworkServiceServer {
+	if newIPAMServer == nil {
+		panic("newIPAMServer should not be nil")
+	}
+	ipPool := ippool.New(net.IPv6len)
```
Comment (denis-tingaikin) [inline pkg/networkservice/ipam/filteripam/server.go:42]: TODO: consider using dualstack ippool

File: pkg/networkservice/ipam/filteripam/server.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	"net/netip"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/ippool"
+)
+
+type filterIPAMServer struct {
```
Comment (denis-tingaikin) [inline pkg/networkservice/ipam/filteripam/server.go:33]: Consider better naming.

File: pkg/networkservice/ipam/filteripam/server.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	ipv6Net := new(net.IPNet)
+	ipv6Net.IP = ipNet.IP.To16()
+	ipv6Net.Mask = make([]byte, 16)
+	copy(ipv6Net.Mask[12:], ipNet.Mask)
+
+	return ipv6Net
+}
+
+func (s *filterIPAMServer) getInvalidAddrs(addrs []string) []string {
+	invalidAddrs := make([]string, 0)
```
Comment (denis-tingaikin) [inline pkg/networkservice/ipam/filteripam/server.go:81]:
```suggestion
	invalidAddrs := []string(nil)
```

File: pkg/networkservice/ipam/filteripam/server.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	for _, addr := range ipContext.DstIpAddrs {
+		_, _ = s.ipPool.PullIPString(addr)
+	}
+}
+
+func (s *filterIPAMServer) free(ipContext *networkservice.IPContext) {
+	for _, addr := range ipContext.SrcIpAddrs {
+		_, ipNet, err := net.ParseCIDR(addr)
+		if err != nil {
+			return
```
Comment (ljkiraly) [inline pkg/networkservice/ipam/filteripam/server.go:141]: We don't expect any error there, but don't we want to free the rest?

**Pull request closed without merging**